cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/KafkaConnectService.java

```diff
@@ -0,0 +1,176 @@
+
+    String suffixUrl =
+        clusterConnectorRequest.getEnv()
+            + "/connectors/"
+            + clusterConnectorRequest.getConnectorName();
+    Pair<String, RestTemplate> reqDetails =
+        clusterApiUtils.getRequestDetails(
+            suffixUrl, clusterConnectorRequest.getProtocol(), KafkaClustersType.KAFKA_CONNECT);
+
+    try {
+      reqDetails.getRight().delete(reqDetails.getLeft(), String.class);
+    } catch (RestClientException e) {
+      log.error("Error in deleting connector " + e.toString());
+      result.put("result", ApiResultStatus.ERROR.value);
+      result.put("errorText", e.toString().replaceAll("\"", ""));
+      return result;
+    }
+    result.put("result", ApiResultStatus.SUCCESS.value);
+    return result;
+  }
+
+  public Map<String, String> updateConnector(ClusterConnectorRequest clusterConnectorRequest) {
+    log.info("Into updateConnector {}", clusterConnectorRequest);
+    Map<String, String> result = new HashMap<>();
+
+    String suffixUrl =
+        clusterConnectorRequest.getEnv()
+            + "/connectors/"
+            + clusterConnectorRequest.getConnectorName()
+            + "/config";
+    Pair<String, RestTemplate> reqDetails =
+        clusterApiUtils.getRequestDetails(
+            suffixUrl, clusterConnectorRequest.getProtocol(), KafkaClustersType.KAFKA_CONNECT);
+
+    HttpHeaders headers = new HttpHeaders();
+    headers.set("Content-Type", "application/json");
+
+    HttpEntity<String> request =
+        new HttpEntity<>(clusterConnectorRequest.getConnectorConfig(), headers);
+
+    try {
+      reqDetails.getRight().put(reqDetails.getLeft(), request, String.class);
+    } catch (RestClientException e) {
+      log.error("Error in updating connector " + e.toString());
+      result.put("result", ApiResultStatus.ERROR.value);
+      result.put("errorText", e.toString().replaceAll("\"", ""));
+      return result;
+    }
+    result.put("result", ApiResultStatus.SUCCESS.value);
+
+    return result;
+  }
+
+  public ApiResponse postNewConnector(ClusterConnectorRequest clusterConnectorRequest)
+      throws Exception {
+    log.info("Into postNewConnector clusterConnectorRequest {} ", clusterConnectorRequest);
+
+    String suffixUrl = clusterConnectorRequest.getEnv() + "/connectors";
+    Pair<String, RestTemplate> reqDetails =
+        clusterApiUtils.getRequestDetails(
+            suffixUrl, clusterConnectorRequest.getProtocol(), KafkaClustersType.KAFKA_CONNECT);
+
+    HttpHeaders headers = new HttpHeaders();
+    headers.set("Content-Type", "application/json");
+
+    HttpEntity<String> request =
+        new HttpEntity<>(clusterConnectorRequest.getConnectorConfig(), headers);
+    ResponseEntity<String> responseNew;
+    try {
+      responseNew =
+          reqDetails.getRight().postForEntity(reqDetails.getLeft(), request, String.class);
+    } catch (RestClientException e) {
+      log.error("Error in registering new connector " + e.toString());
```

Comment:
```suggestion
      log.error("Error in registering new connector ", e);
```
it's better to log the whole stacktrace